src/ride/ride.c

```diff
@@ -995,7 +995,10 @@ void ride_remove_peeps(int rideIndex)
 			exitZ = ride->station_heights[stationIndex];
 			mapElement = ride_get_station_exit_element(ride, exitX, exitY, exitZ);
 
-			exitDirection = mapElement->type & 3;
+			if (mapElement != NULL) {
+				exitDirection = mapElement->type;
+			}
+			exitDirection &= 3;
```

Comment:
I'm slightly concerned adding this fake solution here, as this could hide a bug elsewhere – parks with no rides without exit started popping up recently and we may have a bug that allows such rides, which I don't think is valid. I have a check for that in my other branch, #2581, but it isn't ready yet.

I think I will back this change out and prepare another PR with this only, so we can track it properly.